### ISSUE TITLE
Don't ring on outgoing call

### ISSUE THREAD

**Issue by schiessle** [nextcloud/talk-android#772]
Every time I initiate a call with the Android app the phone rings two or three times like on incoming calls before the call is set up. This is really irritating. I think the phone should only ring on incoming calls but not on outgoing calls.

**Comment by mario** [comment]
1) Absolutely every phone app (also a regular phone as well) does this
2) That being said, this has been brought to the attention of the Talk team and will be discussed how to handle

**Comment by mario** [comment]
Not to mention that this was introduced because people asked for it.

**Comment by schiessle** [comment]
I think the user should be able to disable it / mute it and if we have a acoustic signal it should be a different than for incoming calls (less disruptive, compare it with your normal phone).

**Comment by mario** [comment]
Yes, like I've said earlier - feel free to find me licence-acceptable acoustic signal for this and I'll change it.

**Comment by simonspa** [comment]
>     1. Absolutely every phone app (also a regular phone as well) does this

You are right that there is a sound when calling out - but there should be a difference between a "ring tone" (incoming) and a "dial tone" (outgoing) as for your regular phone.

Otherwise users might be very confused if they pressed "call" and suddenly have an incoming call.

**Comment by mario** [comment]
@simison have you found me a dial tone that's properly licenced yet? :) Ogg format preferable :D

**Comment by simonspa** [comment]
Standard dial tones are available on Wikimedia under CC, but I'm not sure that's exactly what you are looking for.

**Comment by simonspa** [comment]
https://en.m.wikipedia.org/wiki/File:1TR110-1_Kap8.1_Waehlton.ogg

**Comment by mario** [comment]
Will take a look, thanks.

**Comment by ubuntiger** [comment]
> Otherwise users might be very confused if they pressed "call" and suddenly have an incoming call.
I can confirm that I was very confused, especially when initially trying with 2 phones in my hands not knowing how the app has to be used at all.

**Comment by jakobroehrl** [comment]
I think ringing is good, but with another tone. 
And I would be nice, if the tone will stop when the connection is **fully** established. 
Now I have to wait 
1. till the tone is tone and 
2. till I see the avatar in the screen, that is confusing.

**Comment by ericschdt** [comment]
> @simison have you found me a dial tone that's properly licenced yet? :) Ogg format preferable :D

You may have a look at how it was done at [Conersations (Commit: play busy and dial tones)](https://github.com/iNPUTmice/Conversations/commit/fc4397e5b9780a9be6845fe9ad0332829e72cc46)

It uses the `android.media.ToneGenerator` to create harmonic ring tones. :)

**Comment by mahibi** [comment]
i would really like to get rid of the incoming call sound for outgoing calls soon.
so as @mario said: 
a "licence-acceptable acoustic signal preferable in ogg format" would be really great.
How is this done for the iPhone App @Ivansss ? Is there any sound we could use in Android too?

**Comment by simonspa** [comment]
@mahibi above several solutions have been posted:
* use a standard dial tone ogg file from Wikipedia, licensed under CC, in .ogg
* generate dial tone as simple harmonic, linked code as an example

**Comment by Ivansss** [comment]
> How is this done for the iPhone App @Ivansss ? Is there any sound we could use in Android too?

We do not use any sound in the iOS app at the moment. So we could use the same one for both apps when we found one ;)

**Comment by mahibi** [comment]
So just had a closer look at wikipedia fo find "beep beep" tones ;)
@nickvergessen @Ivansss lets pick this one for outgoing calls?: https://commons.wikimedia.org/wiki/File:1TR110-1_Kap8.3_Freiton1.ogg

more can be found here: https://commons.wikimedia.org/wiki/Category:Dial_tones_in_Germany

**Comment by Ivansss** [comment]
> lets pick this one for outgoing calls?: https://commons.wikimedia.org/wiki/File:1TR110-1_Kap8.3_Freiton1.ogg

Sounds good to me :)

**Comment by dvergeylen** [comment]
I know I am late in the game as I see you merged a PR for this but just to advertise one can generate a tone very easily with `speaker-test` from `alsa-utils` package in Debian/Ubuntu (that can later be cropped in Audacity or alikes):

```bash
 speaker-test -t sine -f 440 -l 1
```

This could be a good fallback allowing to generate a Nextcloud owned sound if CC license is incompatible or too restrictive in its allowed usage 🙂.

Thrilled to see talk-android coming back to life! :tada:

**Comment by mahibi** [comment]
thanks @dvergeylen
That would have been a good solution too! 
Anyway i'll keep the ogg file for now.

I will leave this Issues open because the title is "Don't ring on outgoing call" which is not the case yet, but maybe in the future this could be chosen in the settings..

**Comment by mahibi** [comment]
I will keep this issue open because the title is "Don't ring on outgoing call" which is not the case yet. Maybe this should be an option in the settings in the future...

**Comment by nickvergessen** [comment]
Let's close it for now as the current sound is acceptable

**Comment by arnowelzel** [comment]
I just found this issue when trying to disable the ring tone.

Even that issue is already closed and a couple of years old - a suggesiont: The point is, that Nextcloud Talk is **not** a **phone** app. It is a video conferencing client. For example: when I start a meeting in Microsoft Teams or join a meeting in Jitsi Meet, the client won't ring either. In Microsoft Teams the client will only use an outgoing call sound, if I call someone directly, but not when I just join a meeting I created as the first visitor. It is very confusing, when I create a new Talk conversation with multiple other people and join there, that the app rings until some else also joins the conversation.
